templates/pkg/resource/manager.go.tpl

```diff
@@ -157,7 +157,7 @@ func (rm *resourceManager) Delete(
 	if observed != nil {
 		return rm.onSuccess(observed)
 	}
-	return rm.onSuccess(r)
+	return nil, nil
```

Comment:
I think this is not needed anymore because your runtime PR.

`Delete` method should return the AWS resource returned from the AWS delete call.
And the issue that we were patching status on unmanaged resource can be handled in runtime code (alongside the context where we mark the resource unmanaged).

Reply:
Right, that is handled in line 158. This is for the case where resource and error both are returned nil from sdkDelete. With this change, we are returning the output from sdkDelete instead of the AWS resource that was passed to this method. Wdyt?

Reply:
Okay, makes sense & sounds good to me.
Sorry I overlooked.

One suggestion though, how about the code becomes like following. The nil check moves to `rm.onSuccess()` method
```go
...
    return rm.onSuccess(observed)
}

func (rm *resourceManager) onSuccess(
	r *resource,
) (acktypes.AWSResource, error) {
        if ackCompare.isNil(r) {
             return nil, nil
        }
	r1, updated := rm.updateConditions(r, true, nil)
	if !updated {
		return r, nil
	}
	return r1, nil
}
```

Reply:
Good point, this is cleaner and robust. Will make the change

Reply:
Done!